tests/FakeItEasy.Specs/CreationOptionsSpecs.cs

```diff
@@ -835,12 +835,30 @@ public interface IInterfaceThatWeWillAddAttributesTo3
                     .WithArgumentsForConstructor(() => A.Dummy<MakesVirtualCallInConstructor>()));
 
             "When I create a fake using the options builder"
-                .x(() => exception = Record.Exception(() => A.Fake(optionsBuilder)));
+                .x(() => exception = Record.Exception(() => this.CreateFake(optionsBuilder)));
 
             "Then fake creation fails"
                 .x(() => exception.Should().BeAnExceptionOfType<ArgumentException>()
                 .WithMessage("*Only expression of the type ExpressionType.New (constructor calls) are accepted.*"));
         }
+
+        protected abstract T CreateFake<T>(Action<IFakeOptions<T>> optionsBuilder);

```

Comment:
This is the simplest way I found to test both the generic and non-generic overloads. Do you see any issue with it?